### PR TITLE
[Merged by Bors] - Clean up warnings

### PR DESCRIPTION
**Pull request by AgeManning**
This is a small PR that cleans up compiler warnings. 

The most controversial change is removing the `data_dir` field from the `BeaconChainBuilder`. 

It was removed because it was never read.

### CODE REVIEW

File: beacon_node/beacon_chain/src/builder.rs
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/builder.rs:81]: totally in favour of axing this, I noticed that there are a few weird redundant things in here when working on #2234

File: common/deposit_contract/build.rs
Comment (michaelsproul) [inline common/deposit_contract/build.rs]: I think you must be using a Nightly compiler, this API isn't stable: https://doc.rust-lang.org/std/panic/fn.panic_any.html

**Review by michaelsproul — COMMENTED** [review]
I think we should merge this after #2239, as there are some cargo.lock updates I had to do there and there'll be conflicts

File: common/deposit_contract/build.rs
Comment (AgeManning) [reply common/deposit_contract/build.rs]: ah yeah. I forgot I had to switch to nightly to build some solidity tools :|

**Comment by michaelsproul** [comment]
needs a merge to resolve the conflicts

**Comment by AgeManning** [comment]
I've merged the latest unstable. 
It looks like you removed the only other functionality in the `data_dir()` function. 
I have mechanically removed all references to that function. 

We should review to makes sure this is correct and has no side effects.

**Comment by paulhauner** [comment]
I just merged in unstable to fix the conflict.

**Comment by paulhauner** [comment]
@michaelsproul is this related to #2234?

**Review by michaelsproul — APPROVED** [review]
> @michaelsproul is this related to #2234?

Yeah, it seems #2234 made that field completely redundant.

I think this PR is good to merge once all the tests pass.

**Comment by paulhauner** [comment]
@hermanjunge this looks like a pretty odd failure in the remote signer?

**Comment by ghost** [comment]
Pushed cosmetic fixes on two `assert!` cases and one `panic!`. Should do the trick.

@paulhauner @michaelsproul

**Comment by michaelsproul** [comment]
let's merge

bors r+

**Pull request closed without merging**
